### PR TITLE
feat(scripts): add `run-stack` scripts

### PR DESCRIPTION
**Pull request by clemsos**
# Description

This adds a simpler way to run the entire protocol, either dockerized or from local repo.

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Review by searchableguy — APPROVED** [review]

File: .github/workflows/pull-request.yml [hunk 1]
```diff
@@ -24,6 +24,7 @@ jobs:
         "packages/paywall",
         "smart-contracts",
         "subgraph",
+        "tests",
```
Comment (clemsos) [inline .github/workflows/pull-request.yml:27]: this adds the `tests` folder in the CI check so we still run integration tests even if there are changes only in that folder

Comment (clemsos) [reply .github/workflows/pull-request.yml:27]: so that doesnt workas it is  as CI attempts to launch it as unit test

Comment (julien51) [reply .github/workflows/pull-request.yml:27]: We really need a better way to detect changes...

File: tests/package.json [hunk 2]
```diff
@@ -5,7 +5,8 @@
   "scripts": {
     "lint:fix": "yarn lint --fix",
     "lint": "eslint --resolve-plugins-relative-to ../eslint-config --ext .tsx,.ts,.js scripts/ test/",
-    "test": "hardhat test",
+    "test": "yarn test:subgraph",
+    "test:subgraph": "hardhat test",
     "ci": "hardhat test --network 0.0.0.0"
```
Comment (julien51) [inline tests/package.json:10]:
```suggestion
    "ci": "yarn test"
```

File: scripts/run-stack-dockerized.sh [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+# docker-compose $COMPOSE_CONFIG up --build locksmith websub unlock-app 
+
+# Run the integration tests if needed
+if [ "$1" = 'run' ]
+then
+  echo "Running integration tests \n"
+  COMMAND="yarn workspace tests test --network docker"
+  docker-compose $COMPOSE_CONFIG build integration-tests
+  docker-compose $COMPOSE_CONFIG run -e UNLOCK_ENV=test -e CI=true $EXTRA_ARGS integration-tests bash -c "$COMMAND"
+fi
```
Comment (julien51) [inline scripts/run-stack-dockerized.sh:42]:
```suggestion
```

Let's actually remove that so this script is just about setting up the whole stack?

**Review by julien51 — APPROVED** [review]
LG, one minor comment!

File: tests/package.json (hunk 2 above)
Comment (clemsos) [reply tests/package.json:10]: no we need to keep the 0.0.0.0 network here to point towards docker on CI

**Pull request merged**